src/components/Pages/Submit/Form/Tags/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC, useCallback, useEffect, useRef, useState } from "react";
+import { Tag, TagGroup, Input } from "rsuite";
+import { ArticleParams } from "../store/types";
+import { AddLink } from "../../../../Icons";
+
+const removeTag = (
+  tag: string,
+  tags: string[],
+  upsert: (update: ArticleParams) => void
+) => {
+  const nextTags = tags.filter((item) => item !== tag);
+  upsert({ tags: nextTags });
+};
+
+const Tags: FC<{
+  tags: string[];
+  upsert: (update: ArticleParams) => void;
+}> = ({ tags, upsert }) => {
+  const [isEditing, setEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const update = useCallback(
+    (tag: string) => {
+      if (tag == "") {
+        setEditing(false);
+        return;
+      }
+      upsert({ tags: [...tags, tag] });
+    },
+    [tags, setEditing]
+  );
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
```

Comment:
Set editing to false if the article has changed. Will remove the input if the tag is entered or if something else is updated. Technically we can scope this only to tags but this seems like a good place to keep it.